packages/modal/src/ModalView.tsx

```diff
@@ -0,0 +1,300 @@
+import React, { useCallback, useState } from 'react';
+import PropTypes from 'prop-types';
+import FocusTrap from 'focus-trap-react';
+import { Transition } from 'react-transition-group';
+import { transparentize } from 'polished';
+import facepaint from 'facepaint';
+import Portal from '@leafygreen-ui/portal';
+import XIcon from '@leafygreen-ui/icon/dist/X';
+import IconButton from '@leafygreen-ui/icon-button';
+import { useEscapeKey, useIdAllocator } from '@leafygreen-ui/hooks';
+import { palette, uiColors } from '@leafygreen-ui/palette';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { fontFamilies } from '@leafygreen-ui/tokens';
+import {
+  PortalContextProvider,
+  usePopoverContext,
+} from '@leafygreen-ui/leafygreen-provider';
+import { CloseIconColor, ModalProps, ModalSize } from './Modal';
+
+const Mode = {
+  Dark: 'dark',
+  Light: 'light',
+};
+
+type Mode = typeof Mode[keyof typeof Mode];
+
+// breakpoints for different screen sizes
+const small = '767px'; // mobile screens, from 0px - 767px
+const medium = '768px'; // tablet screens, from 768px - 1024px
+const large = '1025px'; // laptops/desktop screens, from 1025px and above
```

Comment:
These aren't even your changes, but could we change this to use `token.breakpoints`? And `palette` in the place of `uiColors` in other spots.

Reply:
I'll hold off and wait till we add dark mode to update all of this